### ISSUE TITLE
Stop (Fade Out) Sounds in Scripts

### ISSUE THREAD

**Issue by k9lego** [daid/EmptyEpsilon#532]
Hello daid, I'll keep my request short.
It would be very helpful for my group if we had fade out (fade speed, maybe??) added to scriptable sounds. A way to end sound files early, but not too abruptly would greatly be appreciated.

It would allow one of my members to configure music based on scriptable cues, such as 10 enemies near the player triggers more intense music than one fighter. While this is technically possible now, sometimes a track needs to be ended early. If this is at all possible, it would be greatly appreciated!! :)
k9lego

**Comment by LJDude** [comment]
I agree. It would be nice to have different stages of music too. For example, you start with one big fleet of ships, and then you shave the fleet down a bit, then the music gets less intense, then shave the fleet down some more, less intense, etc. Or you know, a few ships come in range, then more ships come in range, the music gets more intense. Or, if you have a few ships chasing you (or maybe you're running away from them), and not necessarily attacking you, the music maybe would just be a little tense. Or, maybe instead of in range, the tense music can be if the enemy ships are nearby, and only science can see them.

**Comment by k9lego** [comment]
Yes! We can start sounds/music in scripts already. The only problem is we cannot stop long tracks early. If a long intense music is playing, it cannot be stopped if the enemies are defeated.
Also, a way to know if a sound is still playing certainly would be helpful... :)

_____________________________

**Comment by k9lego** [comment]
@LJDude I was looking into the source code, and there is a "Threat Level" function. That might be useful.
